mllib/src/main/scala/org/apache/spark/ml/fpm/FPGrowth.scala

```diff
@@ -158,18 +159,30 @@ class FPGrowth @Since("2.2.0") (
   }
 
   private def genericFit[T: ClassTag](dataset: Dataset[_]): FPGrowthModel = {
+    val handlePersistence = dataset.storageLevel == StorageLevel.NONE
+
     val data = dataset.select($(itemsCol))
-    val items = data.where(col($(itemsCol)).isNotNull).rdd.map(r => r.getSeq[T](0).toArray)
+    val items = data.where(col($(itemsCol)).isNotNull).rdd.map(r => r.getSeq[Any](0).toArray)
```

Comment:
Is the change from T to Any necessary? I'd presume it does in fact need to be a Seq of the item type.

Reply:
Hm, I wonder how it worked before then? maybe it is related to the caching.
But then this changes the type of `items`, and therefore `parentModel`. Maybe it doesn't matter, just surprising.

T is a ClassTag here and `.toArray` wouldn't work otherwise. Small similar example:

```
scala> def foo[T:scala.reflect.ClassTag](s: Seq[T]) = s.toArray
foo: [T](s: Seq[T])(implicit evidence$1: scala.reflect.ClassTag[T])Array[T]

scala> def foo[T](s: Seq[T]) = s.toArray
<console>:11: error: No ClassTag available for T
       def foo[T](s: Seq[T]) = s.toArray
                                 ^

scala> def foo[T:scala.reflect.ClassTag](s: Seq[T]) = s.toArray
foo: [T](s: Seq[T])(implicit evidence$1: scala.reflect.ClassTag[T])Array[T]

scala> foo(Seq("foo", "bar"))
res3: Array[String] = Array(foo, bar)
```

oh well if it works with this change and not without, seems OK, just an interesting curiosity.

Reply:
It is not only for caching. Same ArrayStoreException occurs if one tries to execute collect() on the items RDD. No exception when using a concrete type like String instead of T. Probably the latter explains how it worked before - people invoked dataset.cache() in their code where type parameter of the Dataset is known.

Reply:
An interesting curiosity for me: why FPGrowth contract requires `Array` of items, not `Seq`? First, it's strange for the contract to require a specific implementation rather than an interface. Second, this leads to redundant `toArray` and back `toSeq` transformations. `Seq` would be more convenient, as `Row` class has `getSeq` method but does not have `getArray`.

Reply:
Yeah, probably a quirk of history. If it works it's fine.